processor/deltatocumulativeprocessor/internal/data/datatest/compare/compare.go

```diff
@@ -14,14 +14,14 @@ import (
 var Opts = []cmp.Option{
 	cmpopts.EquateApprox(0, 1e-9),
 	cmp.Exporter(func(ty reflect.Type) bool {
-		return strings.HasPrefix(ty.PkgPath(), "go.opentelemetry.io/collector/pdata")
+		return strings.HasPrefix(ty.PkgPath(), "go.opentelemetry.io/collector/pdata") || strings.HasPrefix(ty.PkgPath(), "github.com/open-telemetry/opentelemetry-collector-contrib")
```

Comment:
What are we trying to accomplish here? I'm struggling to understand how this package is being used 😬

Reply:
`compare` is mostly a diffing helper for `pdata`/`pmetric` datatypes.
During testing, the `testify/assert` output was hardly readable, so I switched to `google/go-cmp`.
The latter however refuses to operate on private fields by default however, making it totally unusable for otel datatypes.
above is a simple whitelist for types we know shall be diffed on their private fields, to avoid having to type that in every single test.